### PR TITLE
Migrate away from allensdk

### LINKED ISSUE #67: Remove AllenSDK dependency
[AllenSDK cannot be installed with Python 3.12](https://github.com/AllenInstitute/AllenSDK/issues/2744). As good as this package is, it seems that we can't always rely on it to be maintained satisfactorily. I also think it's too large and complex for us to be able to contribute meaningfully to it's maintenance (although I could be wrong).

To get around this, we should remove the dependency and fetch the data directly (e.g. via REST API). We could also fetch the streamlines data this way, so we no longer rely on neuromorpho.org. 

cc @IgorTatarnikov

### PR DESCRIPTION
**Pull request by IgorTatarnikov**
Before submitting a pull request (PR), please read the [contributing guide](https://github.com/brainglobe/.github/blob/main/CONTRIBUTING.md).

Please fill out as much of this template as you can, but if you have any problems or questions, just leave a comment and we will help out :)

## Description

**What is this PR**

- [ ] Bug fix
- [ ] Addition of a new feature
- [x] Other

**Why is this PR needed?**
We want to remove allensdk as a dependency to make upgrading to python 3.12 easy, see [here](https://github.com/AllenInstitute/AllenSDK/issues/2744)

**What does this PR do?**
Uses the Allen web API to fetch the reconstructed neurons instead of using the allensdk.

## References
Closes #67 

## How has this PR been tested?
All tests pass, I've also tested it locally to make sure we can fetch a variety of files.

## Is this a breaking change?
No, I made sure that the DataFrame stored in the `neurons` attribute of `AllenMorphology` class has the exact same format.

## Does this PR require an update to the documentation?
No

## Checklist:

- [x] The code has been tested locally
- [ ] Tests have been added to cover all new functionality (unit & integration)
- [ ] The documentation has been updated to reflect any changes
- [x] The code has been formatted with [pre-commit](https://pre-commit.com/)

### CODE REVIEW

**Comment by IgorTatarnikov** [comment]
Tests are currently failing due to neuromorpho being down :disappointed:

File: morphapi/api/allenmorphology.py [hunk 1]
```diff
@@ -58,13 +63,92 @@ def __init__(self, *args, **kwargs):
 
         self.downloaded_neurons = self.get_downloaded_neurons()
 
+    def get_cells(self, require_reconstruction: bool = True) -> pd.DataFrame:
+        """
+        Download the metadata for all neurons in the Allen database and save
+        it to a cells.json file.
+        """
+        cells_path = Path(
```
Comment (adamltyson) [inline morphapi/api/allenmorphology.py:71]: Can this use Pathlib rather than os.path?

**Review by adamltyson — APPROVED** [review]
LGTM, thanks!

File: morphapi/api/allenmorphology.py (hunk 1 above)
Comment (IgorTatarnikov) [reply morphapi/api/allenmorphology.py]: Definitely, done now!

**Pull request merged**